source/slang/slang-emit.cpp

```diff
@@ -273,6 +274,18 @@ Result linkAndOptimizeIR(
     if (!compileRequest->allowDynamicCode)
         specializeModule(irModule);
 
+    switch (target)
+    {
+    case CodeGenTarget::CPPSource:
+        // For targets that supports dynamic dispatch, we need to lower the
+        // generics / interface types to ordinary functions and types using
+        // function pointers.
+        lowerGenerics(irModule);
```

Comment:
Is `lowerGenerics` a sufficiently descriptive name? I can't think of something strictly better.

Something like `lowerGenericFunctionsToOrdinaryFunctions` is more explicit about what happens right now, but ignores the fact that the pass is likely to also handle lowering of generic types, witness tables, etc. in the long run.